pyuvdata/uvdata/uvdata.py

```diff
@@ -5674,6 +5702,88 @@ def resample_in_time(self, target_time, only_downsample=False, only_upsample=Fal
+
+        Parameters
+        ----------
+        n_chan_to_avg : int
+            Number of channels to average together.
+        summing_correlator_mode : bool
+            Option to integrate or split the flux from the original samples
+            rather than average or duplicate the flux from the original samples
+            to emulate the behavior in some correlators (e.g. HERA).
+        """
+        self._check_freq_spacing()
+
+        n_final_chan = int(np.floor(self.Nfreqs / n_chan_to_avg))
+        nfreq_mod_navg = self.Nfreqs % n_chan_to_avg
+        if nfreq_mod_navg != 0:
+            # not an even number of final channels
+            warnings.warn("Nfreqs does not divide by `n_chan_to_avg` evenly. "
+                          "The final {} frequencies will be excluded, to "
+                          "control which frequencies to exclude, use a "
+                          "select to control.".format(nfreq_mod_navg))
+            chan_to_keep = np.arange(n_final_chan * n_chan_to_avg)
+            self.select(freq_chans=chan_to_keep)
+
+        self.freq_array = self.freq_array.reshape((self.Nspws, n_final_chan,
+                                                   n_chan_to_avg)).mean(axis=2)
+        self.channel_width = self.channel_width * n_chan_to_avg
+        self.Nfreqs = n_final_chan
+
+        if self.eq_coeffs is not None:
+            eq_coeff_diff = np.diff(self.eq_coeffs, axis=1)
+            if np.abs(np.max(eq_coeff_diff)) > 0:
+                warnings.warn("eq_coeffs vary by frequency. They should be "
+                              "applied to the data using `remove_eq_coeffs` "
+                              "before frequency averaging.")
+            self.eq_coeffs = self.eq_coeffs.reshape((self.Nants_telescope,
+                                                     n_final_chan,
+                                                     n_chan_to_avg)).mean(axis=2)
+
+        if not self.metadata_only:
+            shape_tuple = (self.Nblts, self.Nspws, n_final_chan, n_chan_to_avg,
+                           self.Npols)
+
+            mask = self.flag_array.reshape(shape_tuple)
+
+            # if all inputs are flagged, the flag array should be True,
+            # otherwise it should be False.
+            # The sum below will be zero if it's all flagged and
+            # greater than zero otherwise
+            # Then we use a test against 0 to turn it into a Boolean
+            self.flag_array = np.sum(~self.flag_array.reshape(shape_tuple), axis=3) == 0
+
+            # need to update mask if a downsampled visibility will be flagged
+            # so that we don't set it to zero
+            for n_chan in np.arange(n_final_chan):
+                if (self.flag_array[:, :, n_chan]).any():
+                    ax0_inds, ax1_inds, ax3_inds, ax4_inds = np.nonzero(mask[:, :, n_chan, :])
+                    mask[ax0_inds, ax1_inds, :, ax3_inds, ax4_inds] = False
```

Comment:
Shouldn't the colon correspond to the `n_chan_to_avg` axis (right now it's the `n_final_chan` axis)?

Reply:
oh dear. I'll add a test to catch this and then fix it.

Reply:
oh! it's actually ok, but instead of a colon I could have used a zero because we're iterating over that axis, so that axis is only length 1.

Reply:
(There is already a test that catches that but I made it even more clear)

Reply:
This was actually a bug, just pushed a fix.